am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -63,40 +87,57 @@ void returnListOfPermissionsForRoleName() {
             RESOURCE_TYPE, RESOURCE_NAME, ROLE_NAMES);
 
         assertThat(accessRecord).hasSize(3);
-        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/test"), READ_PERMISSION);
-        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/test2"), READ_PERMISSION);
-        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/testCreate"), CREATE_PERMISSION);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/child"), READ_PERMISSION);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/parent/age"), CREATE_PERMISSION);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/address/street/line1"), CREATE_PERMISSION);
     }
 
     @Test
     void shouldReturnNullWhenServiceNameDoesNotExist() {
         Map<JsonPointer, Set<Permission>> accessRecord = service.getRolePermissions("Unknown Service",
             RESOURCE_TYPE, RESOURCE_NAME, ROLE_NAMES);
 
-        assertThat(accessRecord).isNull();
+        assertThat(accessRecord).isEmpty();
     }
 
     @Test
     void shouldReturnNullWhenResourceTypeDoesNotExist() {
         Map<JsonPointer, Set<Permission>> accessRecord = service.getRolePermissions(SERVICE_NAME,
-            "Unknown Resource Type ", RESOURCE_NAME, ROLE_NAMES);
+            "Unknown Resource Type", RESOURCE_NAME, ROLE_NAMES);
 
-        assertThat(accessRecord).isNull();
+        assertThat(accessRecord).isEmpty();
     }
 
     @Test
     void shouldReturnNullWhenResourceNameDoesNotExist() {
         Map<JsonPointer, Set<Permission>> accessRecord = service.getRolePermissions(SERVICE_NAME,
             RESOURCE_TYPE, "Unknown Resource Name", ROLE_NAMES);
 
-        assertThat(accessRecord).isNull();
+        assertThat(accessRecord).isEmpty();
     }
 
     @Test
     void shouldReturnNullWhenDefaultRoleNameDoesNotExist() {
         Map<JsonPointer, Set<Permission>> accessRecord = service.getRolePermissions(SERVICE_NAME,
             RESOURCE_TYPE, RESOURCE_NAME, Stream.of("Unknown Role").collect(toSet()));
 
-        assertThat(accessRecord).isNull();
+        assertThat(accessRecord).isEmpty();
+    }
+
+    @Test
+    void shouldMergeDataAsExpectedWhenRetrievingPermissionsForMultipleRoles() {
+        Set<String> userRoles = ImmutableSet.of(ROLE_NAME, OTHER_ROLE_NAME);
+        Set<Permission> readUpdate = Stream.of(Permission.READ, Permission.UPDATE).collect(toSet());
+        Set<Permission> createUpdate = Stream.of(Permission.CREATE, Permission.UPDATE).collect(toSet());
+
+        Map<JsonPointer, Set<Permission>> accessRecord = service.getRolePermissions(SERVICE_NAME,
+            RESOURCE_TYPE, RESOURCE_NAME, userRoles);
+
+        assertThat(accessRecord).hasSize(5);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf(""), UPDATE_PERMISSION);
```

Comment:
Can we avoid repeating `assertThat`? What about:

```
assertThat(accessRecord)
  .hasSize(5)
  .containsEntry(JsonPointer.valueOf(""), UPDATE_PERMISSION)
  .containsEntry(JsonPointer.valueOf("/address"), createUpdate)
```

Reply:
Really good spot, I didn't know you tag muliple conditions to one assert 👍 

Changed in another place, there are almost definitely other places this could be done also